components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreAdmin.java

```diff
@@ -66,151 +72,203 @@
-                    names[count] = ite.next();
-                    count++;
                 }
+            }
 
-                if (isSuperTenant) {
-                    KeyStoreData data = new KeyStoreData();
-                    ServerConfiguration config = ServerConfiguration.getInstance();
-                    String fileName = config
-                            .getFirstProperty(RegistryResources.SecurityManagement.SERVER_PRIMARY_KEYSTORE_FILE);
-                    String type = config
-                            .getFirstProperty(RegistryResources.SecurityManagement.SERVER_PRIMARY_KEYSTORE_TYPE);
-                    String name = KeyStoreUtil.getKeyStoreFileName(fileName);
-                    data.setKeyStoreName(name);
-                    data.setKeyStoreType(type);
-                    data.setProvider(" ");
-                    data.setPrivateStore(true);
-
-                    names[count] = data;
-                }
+            // Prepare the next position for the super tenant keystore data.
+            names = new KeyStoreData[lst.size() + 1];
+            Iterator<KeyStoreData> ite = lst.iterator();
+            int count = 0;
+            while (ite.hasNext()) {
+                names[count] = ite.next();
+                count++;
+            }
+
+            if (isSuperTenant) {
+                KeyStoreData data = new KeyStoreData();
+                ServerConfiguration config = ServerConfiguration.getInstance();
+                String fileName = config
+                        .getFirstProperty(RegistryResources.SecurityManagement.SERVER_PRIMARY_KEYSTORE_FILE);
+                String type = config
+                        .getFirstProperty(RegistryResources.SecurityManagement.SERVER_PRIMARY_KEYSTORE_TYPE);
+                String name = KeyStoreUtil.getKeyStoreFileName(fileName);
+                data.setKeyStoreName(name);
+                data.setKeyStoreType(type);
+                data.setProvider(" ");
+                data.setPrivateStore(true);
 
+                names[count] = data;
             }
             return names;
-        } catch (RegistryException e) {
+        } catch (KeyStoreManagementException e) {
+            // Catch KeyStoreManagementException and throw the expected SecurityConfigException.
             String msg = "Error when getting keyStore data";
             log.error(msg, e);
             throw new SecurityConfigException(msg, e);
         }
     }
 
+    /**
+     * Method to add keystore when a file path is given instead of file data.
+     *
+     * @param filePath   - File path of the keystore data.
+     * @param filename   - Name of the keystore.
+     * @param password   - Password of the keystore.
+     * @param provider   - Provider of the keystore.
+     * @param type       - Type of the keystore.
+     * @param pvtKeyPass - Password of the private key.
+     * @throws SecurityConfigException - Throws if an error occurred while adding the keystore.
+     */
```

Comment:
```suggestion
    /**
     * Method to add keystore when a file path is given instead of file data.
     *
     * @param filePath    File path of the keystore data.
     * @param filename    Name of the keystore.
     * @param password    Password of the keystore.
     * @param provider    Provider of the keystore.
     * @param type        Type of the keystore.
     * @param pvtKeyPass  Password of the private key.
     * @throws SecurityConfigException  Throws if an error occurred while adding the keystore.
     */
```
check other places as well

Reply:
Fixed with 6dbdf6fe392c781def1c225d41e87e3549bc36d0